File: dailyQuestion/2020-07/07-23/golang/solution.go
func isUgly(num int) bool {
    if num < 1 {
        return false
    }
    if num < 7 {
        return true
    }

    if num % 2 == 0 {
        num /= 2
    } else if num % 3==0 {
        num /= 3
    } else if num % 5 == 0 {
        num /= 5
    } else {
        return false
    }
    return isUgly(num)
}

File: dailyQuestion/2020-07/07-23/python/solution.py
def isUgly(num: int) -> bool:
    if num < 1 :
        return False
    if num < 7 :
        return True
    if num % 2 == 0 :
        num /= 2
    elif num % 3 == 0 :
        num /= 3
    elif num % 5 == 0 :
        num /= 5
    else :
        return False
    return isUgly(num)

if __name__ == "__main__" :
    num = 34
    s = isUgly(num)
    print(s)